libraries/ESP8266WiFi/src/WiFiClient.cpp

```diff
@@ -256,7 +256,7 @@ int WiFiClient::available()
     int result = _client->getSize();
 
     if (!result) {
-        optimistic_yield(100);
+        optimistic_yield(10000);
```

Comment:
Here the user has asked if there's more data and the answer is no.  There can't be more data until a yield() happens, so by increasing the delay I think you'll decrease performance.